src/Examples/nip17-private-direct-messages.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+declare(strict_types=1);
+
+use swentel\nostr\Key\Key;
+use swentel\nostr\Nip17\DirectMessage;
+use swentel\nostr\Nip59\GiftWrapService;
+use swentel\nostr\Sign\Sign;
+
+require __DIR__ . '/../../vendor/autoload.php';
+
+// Set up keys for Alice and Bob
+$key = new Key();
+$sign = new Sign();
+$giftWrapService = new GiftWrapService($key, $sign);
+
+try {
+    // Alice's private and public keys
+    $alicePrivKey = '0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef';
+    $alicePubKey = $key->getPublicKey($alicePrivKey);
+    print "Alice's public key: " . $alicePubKey . PHP_EOL;
+
+    // Bob's private and public keys
+    $bobPrivKey = 'fedcba9876543210fedcba9876543210fedcba9876543210fedcba9876543210';
+    $bobPubKey = $key->getPublicKey($bobPrivKey);
+    print "Bob's public key: " . $bobPubKey . PHP_EOL;
+
+    // Create DirectMessage service
+    $directMessage = new DirectMessage($giftWrapService, $key);
+
+    print PHP_EOL . "===== SENDING A PRIVATE DIRECT MESSAGE =====" . PHP_EOL;
+
+    // Message from Alice to Bob
+    $messageText = "Hey Bob, this is a private message that only you can read!";
+    print "Alice is sending message to Bob: '" . $messageText . "'" . PHP_EOL;
+
+    // Send the direct message, creating gift wraps for both sender and receiver
+    $result = $directMessage->sendDirectMessage(
+        $alicePrivKey,
+        $bobPubKey,
+        $messageText,
+    );
+
+    // Display information about the created gift wraps
+    print PHP_EOL . "Two gift wraps were created:" . PHP_EOL;
+    print "1. Receiver gift wrap (for Bob):" . PHP_EOL;
+    print "   - Kind: " . $result['receiver']->getKind() . PHP_EOL;
+    print "   - ID: " . $result['receiver']->getId() . PHP_EOL;
+
+    print PHP_EOL . "2. Sender gift wrap (for Alice):" . PHP_EOL;
+    print "   - Kind: " . $result['sender']->getKind() . PHP_EOL;
+    print "   - ID: " . $result['sender']->getId() . PHP_EOL;
+
+    print PHP_EOL . "===== PUBLISHING GIFT WRAPS TO RELAYS =====" . PHP_EOL;
+
+    // Normally, you would get the relay list from the NIP-10050 event
```

Comment:
Do you mean NIP-65 here with event kind `10002`?

Reply:
Oops... I meant kind 10050, as described in [NIP-51 ](https://github.com/nostr-protocol/nips/blob/master/51.md)